File: java/medium/55_jump_game.java
package java.medium;

class JumpGame {
    public boolean canJump(int[] nums) {
        int n = nums.length;
        int lastPos = n;

        for (int i = n - 1; i >= 0; i--) {
            if (i+ nums[i] >= lastPos) {
                lastPos = i;
            }
        }

        return lastPos == 0;
    }
}

File: go/medium/55.jump_game.go
// https://leetcode.com/problems/jump-game
package medium

func CanJump(nums []int) bool {
	lastPost := len(nums) - 1
	for i := len(nums) - 1; i >= 0; i-- {
		if i+nums[i] >= lastPost {
			lastPost = i
		}
	}
	return lastPost == 0
}
